### ISSUE TITLE
Block until server i18n instance detects a language

### ISSUE THREAD

**Issue by dhalbrook** [isaachinman/next-i18next#583]
When I enable debugging, I notice that in the SSR pass, rendering begins before a language has been detected.  However, there is a proper await on the client pass, so the React side is not re-rendered w/o a language.

This is happening in 2.1.1 (latest i think).

One way to verify this is to output the current language of the i18Next.i18n object in your components as they render, and also in the middleware.

I'd expect the SSR pass to block until the language is detected.  Perhaps next-i18nextmiddleware.ts can provide an option in the config to pass it a callback function to call when the language is detected, but maybe that means modifying i18next-express-middleware as well.  

Any ideas are appreciated.

**Comment by isaachinman** [comment]
> One way to verify this is to output the current language of the i18Next.i18n

No, that's not accurate. During SSR, `req.i18n` is what's used for localisation.

Not sure if this issue is a misunderstanding, or a genuine problem. Can you help me understand a bit more? Perhaps an example repo with some helpful console logging in place?

**Comment by dhalbrook** [comment]
Yes, I may be referencing the wrong 1i8n instance.  Let me verify that and that's not the case, i'll work on getting a sample repo together.   Thx for the feedback.

**Comment by dhalbrook** [comment]
So I've been looking at this a bit more and it's pretty easy to reproduce.  It happens both in dev mode and in a full production build.

I shimmed the `useTranslation` hook and added some logic like this:

  ```javascript
const resp = nextI18Next.useTranslation(namespace, options);
  const { i18n } = resp;
  const { language } = i18n;
  console.log('>> i18n.language for ns:', namespace, language);
```

on the server pass, I seem to be getting multiple render passes, where the first one does not have a language associated with the `i18n` object returned from the hook:

```
>> i18n.language for ns: pages_community_CommunityPage undefined
>> i18n.language for ns: components_community_NavbarWidget undefined
>> i18n.language for ns: components_search_SpotlightSearchIcon undefined
```
then on another pass, then language is set:

```
>> i18n.language for ns: pages_community_CommunityPage en
>> i18n.language for ns: components_community_NavbarWidget en
>> i18n.language for ns: components_search_SpotlightSearchIcon en
```
on the client, there is only a single pass corresponding to the latter set of log statements.  I'm wondering if the language setting event is triggering the second render pass on the server.

My `server.js` is fairly straightforward:

```
const express = require('express');
const next = require('next');
const nextI18NextMiddleware = require('next-i18next/middleware').default;
const { router } = require('./routes');
const nextI18next = require('./utils/i18n');

const port = process.env.PORT || 3000;
const app = next({ dev: process.env.NODE_ENV !== 'production' });
const handler = router.getRequestHandler(app);

(async () => {
  await app.prepare();
  const server = express();
  server.use(nextI18NextMiddleware(nextI18next));
  server.use(handler);
  await server.listen(port);
  console.log(`> Ready on http://localhost:${port}`); // eslint-disable-line no-console
})();
```

Our `i18n.js` file just initializes a `next18next` instance.

Any ideas are appreciated.  LMK if you still need a sample repo or if this is enough.

**Comment by isaachinman** [comment]
Thanks for looking into it further. Sounds like we've got a genuine issue here. A demo repo would be great - I will use that to debug and fix.

**Comment by dhalbrook** [comment]
Just a quick update... I've been trying to repro using the sample app and unable to do so, so it's something specific to our usage and not necessarily anything in the project itself.  I'll keep you posted on where it leads.

**Comment by dhalbrook** [comment]
I've narrowed down the double-pass to https://github.com/zeit/next.js/blob/canary/examples/with-apollo/lib/apollo.js

`export default withApollo(LangTest);`

I created a page in our app that simply outputs the language and logs it.  Normally it works as expected.  When I wrap it with the HoC, it does the double pass, during which the first pass the language is undefined, even though there's no Apollo query executing.
